src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKStyle.java

```diff
@@ -182,6 +182,30 @@ else if (type == GTKColorType.WHITE) {
             }
         }
 
+        if (id == Region.LABEL && type == ColorType.FOREGROUND
+                && (state & SynthConstants.ENABLED) != 0) {
+            Color c = (Color) get(context, "Label.foreground");
+            if (c != null) {
+                return c;
+            }
+        }
+
+        if (id == Region.CHECK_BOX && type == ColorType.FOREGROUND
+                && (state & SynthConstants.DISABLED) != 0) {
+            Color c = (Color) get(context, "CheckBox.disabledText");
+            if (c != null) {
+                return c;
+            }
+        }
+
+        if (id == Region.RADIO_BUTTON && type == ColorType.FOREGROUND
+                && (state & SynthConstants.DISABLED) != 0) {
+            Color c = (Color) get(context, "RadioButton.disabledText");
+            if (c != null) {
+                return c;
```

Comment:
What about the enabled state text? WIll that not required to be handled? 
Can this be done in a separate method passing in the required Java level property?

Reply:
> What about the enabled state text? WIll that not required to be handled?
If the component is in enabled state and color is set by the user then it needs to handle as well. Since the bug reported was for disabled state, didn't add for enabled as of now.

> Can this be done in a separate method passing in the required Java level property?
Yeah sure. Will check and update.